src/xToken.ts

```diff
@@ -732,4 +738,27 @@ export class XToken {
         return mintXU3LP(symbol, tradeWithEth ? 1 : 0, value, this.provider)
     }
   }
+
+  public async stakeXtk(amount: string) {
+    return stakeXtk(this.provider, amount)
```

Comment:
Is the `amount` being passed exactly what user inputs in the UI or it's being parsed as ether value i.e., `10 ** 18`?
If it's being passed as user input, the `BigNumber.from(amount)` will throw an error for decimals. If it's being passed as `10 ** 18` value then you should really convert it to be the same as user input value by leveraging `formatEther()` and `parseEther()` methods. Refer to other usages we already have, for example, mint/burn methods.